unified-scheduler-logic/src/lib.rs

```diff
@@ -911,19 +931,31 @@ impl SchedulingStateMachine {
     /// # Safety
     /// Call this exactly once for each thread. See [`TokenCell`] for details.
     #[must_use]
-    pub unsafe fn exclusively_initialize_current_thread_for_scheduling() -> Self {
+    pub unsafe fn exclusively_initialize_current_thread_for_scheduling(
+        max_executing_task_count: Option<usize>,
```

Comment:
why pass `usize` instead of `u32`?

Reply:
thanks for spotting some obscure choice of api in this regard.. hope this commit helps my intention: 17d5a0bccb0ce948f8e6838c6cf70fc3a73885d0